mrmustard/math/lattice/strategies/compactFock/inputValidation.py

```diff
@@ -85,3 +85,22 @@ def grad_hermite_multidimensional_1leftoverMode(A, B, G0, arr0, arr2, arr1010, a
     )
     arr0_dG0 = np.array(arr0 / G0).astype(np.complex128)
     return arr0_dG0, arr0_dA, arr0_dB
+
+
+def hermite_multidimensional_diagonal_batch(A, B, G0, cutoffs, rtol=1e-05, atol=1e-08):
```

Comment:
are you sure that you want to validate?
validation increases the runtime. If eith wrong outputs the code fails, I would suggest replacing this with try .. except, and the except has a nice message that says what kind of input is expected

Reply:
Good point, I'll go for a new PR to fix this file.